source/MaterialXRender/Handlers/TestObjLoader.cpp

```diff
@@ -248,29 +276,32 @@ bool TestObjLoader::load(const std::string& fileName, MeshList& meshList)
         normalData.push_back(norm[vertexIndex + 2]);
     }
 
-    // Set up flattened indexing
-    MeshPartitionPtr partition = nullptr;
-    if (faceCount && positionData.size() && pidx.size())
+    // Set up flattened contiguous indexing for all partitions
+    unsigned int startIndex = 0;
+    for (auto partition : partitions)
     {
-        partition = MeshPartition::create();
-        partition->setFaceCount(faceCount);
         MeshIndexBuffer& indexing = partition->getIndices();
-        indexing.resize(pidx.size());
-        std::iota(indexing.begin(), indexing.end(), 0);
+        size_t indexCount = partition->getFaceCount() * 3;
+        if (indexCount > 0)
+        {
+            indexing.resize(indexCount);
+            std::iota(indexing.begin(), indexing.end(), startIndex);
+            startIndex += (unsigned int)indexing.size();
```

Comment:
Previous code has 1 contiguous array from 0 -> vertex count*3. Now it's split across each partition. The number stays the same since it's just N partitions accessing a single set of streams vs 1 partition.